### PR TITLE
Sign package release metadata at publication time

### PR DESCRIPTION
**Pull request by yim-lee**
Implement changes as proposed in amendent to SE-0391: https://github.com/apple/swift-evolution/pull/1977

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

File: Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift [hunk 1]
```diff
@@ -191,17 +181,34 @@ extension SwiftPackageRegistryTool {
...
                 swiftTool.observabilityScope.emit(info: "signing the archive at '\(archivePath)'")
-                let signaturePath = workingDirectory
+                contentPaths.append(archivePath)
+                contentSignaturePaths[archivePath] = workingDirectory
                     .appending("\(self.packageIdentity)-\(self.packageVersion).sig")
-                signature = try PackageArchiveSigner.sign(
-                    archivePath: archivePath,
-                    signaturePath: signaturePath,
+
+                if let metadataPath = self.customMetadataPath {
```
Comment (tomerd) [inline Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift:192]: does this mean the user must specify the metadata path, or is there still a default location (ie <root>/package-metadata.json) where we look for it if not specified?

**Review by tomerd — APPROVED** [review]
nice. one question

File: Tests/PackageRegistryTests/RegistryClientTests.swift [hunk 2]
```diff
@@ -1700,8 +1700,8 @@ final class RegistryClientTests: XCTestCase {
             skipSignatureValidation: false,
             signingEntityStorage: .none,
             signingEntityCheckingMode: .strict,
-            customHTTPClient: httpClient
-            ,delegate: .none
+            customHTTPClient: httpClient,
```
Comment (tomerd) [inline Tests/PackageRegistryTests/RegistryClientTests.swift:1703]: thanks

File: Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/PackageRegistryTool/PackageRegistryTool+Publish.swift:192]: yes, user must specify path if they want to include metadata, and metadata is no longer required.

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Review by neonichu — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**